core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -338,7 +338,41 @@ object Chunk {
       }
 
   /** Creates a chunk backed by a `Seq`. */
-  def seq[O](s: Seq[O]): Chunk[O] = vector(s.toVector)
+  def seq[O](s: Seq[O]): Chunk[O] = s match {
+    case a: collection.mutable.WrappedArray[O] => array(a.array)
+    case v: Vector[O]                          => vector(v)
+    case ix: IndexedSeq[O]                     => indexedSeq(ix)
+    case _                                     => buffer(collection.mutable.Buffer(s: _*))
```

Comment:
Note we can't convert the seq to an array here b/c we don't have a `ClassTag[O]`.